source/renderer/app/components/hardware-wallet/settings/ConnectHardwareWallet.js

```diff
@@ -0,0 +1,154 @@
+import trezorIcon from '../../../assets/images/hardware-wallet/trezor-ledger.inline.svg';
+import exportIcon from '../../../assets/images/hardware-wallet/export.inline.svg';
+import checkIcon from '../../../assets/images/hardware-wallet/check.inline.svg';
+import clearIcon from '../../../assets/images/hardware-wallet/close-cross-red.inline.svg';
+import ledgerSmallIcon from '../../../assets/images/hardware-wallet/ledger-bold-ic.inline.svg';
+import styles from './ConnectHardwareWallet.scss';
+import LoadingSpinner from '../../widgets/LoadingSpinner';
+
+const messages = defineMessages({
+  hardwareWalletTitle: {
+    id: 'wallet.hardware.hardwareWalletTitle',
+    defaultMessage: '!!!Hardware wallet',
+    description: 'Hardware wallet title.',
+  },
+  ledgerWalletTitle: {
+    id: 'wallet.hardware.ledgerWalletTitle',
+    defaultMessage: '!!!Ledger wallet',
+    description: 'Ledger wallet title.',
+  },
+  hardwareWalletInstructions: {
+    id: 'wallet.hardware.hardwareWalletInstructions',
+    defaultMessage: '!!!Follow instructions to access your wallet',
+    description: 'Follow instructions label',
+  },
+  hardwareWalletLedgerBegin: {
+    id: 'wallet.hardware.hardwareWalletLedgerBegin',
+    defaultMessage:
+      '!!!To begin, connect and unlock your <span>Ledger Device</span>',
+    description: 'Connect device label',
+  },
+  hardwareWalletBegin: {
+    id: 'wallet.hardware.hardwareWalletBegin',
+    defaultMessage:
+      '!!!To begin, connect and unlock your <span>Hardware wallet Device</span>',
+    description: 'Connect device label',
+  },
+  hardwareWalletExport: {
+    id: 'wallet.hardware.hardwareWalletExport',
+    defaultMessage: '!!!Export <span>public key</span> on your device',
+    description: 'Export wallet label',
+  },
+});
+
+type Props = {
+  onOpenExternalLink: Function,
+  isLedger: boolean,
+  isTrezor: boolean,
+  isDeviceConnected: boolean | null,
+  fetchingDevice: boolean,
+  exportingExtendedPublicKey: boolean | null,
+  isExportingPublicKeyAborted: boolean,
+};
+
+@observer
+export default class ConnectHardwareWallet extends Component<Props> {
+  static contextTypes = {
+    intl: intlShape.isRequired,
+  };
+
+  render() {
+    const { intl } = this.context;
+
+    const link = '';
+
+    const {
+      onOpenExternalLink,
+      isLedger,
+      isTrezor,
+      isDeviceConnected,
+      fetchingDevice,
+      exportingExtendedPublicKey,
+      isExportingPublicKeyAborted,
+    } = this.props;
+
+    const hardwareTitle = isTrezor
+      ? intl.formatMessage(messages.hardwareWalletTitle)
+      : intl.formatMessage(messages.ledgerWalletTitle);
+
+    const hardwareConnectLabel = isTrezor
+      ? messages.hardwareWalletBegin
+      : messages.hardwareWalletLedgerBegin;
+
+    return (
+      <>
+        <div className={styles.component}>
+          <div className={styles.hardwareWalletContainer}>
+            <div className={styles.hardwareWalletWrapper}>
```

Comment:
@DeeJayElly is this necessary? We have 3 div' here for one content inside. Can we wrap content in one div and without Fragment wrapper?

Reply:
Correct, fixed